valohai_yaml/objs/pipelines/execution_node.py

```diff
@@ -33,8 +33,20 @@ def lint(self, lint_result: LintResult, context: LintContext) -> None:
         super().lint(lint_result, context)
         config = context['config']
         pipeline = context['pipeline']
-        if self.step not in config.steps:
-            lint_result.add_error(f'Pipeline {pipeline.name} node {self.name} step {self.step} does not exist')
+        step = config.steps.get(self.step)
+        error_prefix = f'Pipeline {pipeline.name} node {self.name} step {self.step}'
+        if not step:
+            lint_result.add_error(f'{error_prefix} does not exist')
+            return
+        step_inputs = step.inputs.keys()
+        step_parameters = step.parameters.keys()
+        if self.override is not None:
+            for input_name in self.override.inputs:
```

Comment:
could've defined these parameters inside the condition but not a minor detail :detective: